Core/NoOpRunner.Core/Shapes/PowerUp.cs

```diff
@@ -0,0 +1,66 @@
+﻿using NoOpRunner.Core.Entities;
+using NoOpRunner.Core.Enums;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NoOpRunner.Core.Shapes
+{
+    public class PowerUp : BaseShape
+    {
+        // Player 1 PowerUp | Player 2 PowerUp
+        Dictionary<string, string> powerUps = new Dictionary<string, string>(){
+            { "speed_boost", "rocket" },
+            { "invisibility", "proximity_mine" },
+            { "invulnerablity", "saw" },
+            { "double_jump", "knockback_bomb" }
+        };
+
+        public PowerUp(int centerPosX, int centerPosY, int[] platformXCoords, int[] platformYCoords) : base(centerPosX, centerPosY)
```

Comment:
The responsibility to find where to place the powerup shouldn't be of this class. It should only receive exact coordinates of where to render and that's it.